File: TypeScript/Codewars/src/resolutions/Esolang-Interpreters-1-Introduction-to-Esolangs-and-My-First-Interpreter-MiniStringFuck.ts
export function myFirstInterpreter(code: string): string {
	const output: string[] = [];
	let memory_cell = 0;

	for (const element of code) {
		if (element === "+") {
			memory_cell = (memory_cell + 1) % 256;
		} else if (element === ".") {
			output.push(String.fromCharCode(memory_cell));
		}
	}

	return output.join("");
}

File: Csharp/CodeWars/src/Completed/EsolangInterpreters1IntroductionToEsolangsAndMyFirstInterpreterMinistringfuck.cs
namespace CodeWars.Completed;

public class EsolangInterpreters1IntroductionToEsolangsAndMyFirstInterpreterMinistringfuck
{
    public static string MyFirstInterpreter(string code)
    {
        List<char> output = [];
        int memoryCell = 0;

        foreach (var c in code)
        {
            if (c == '+')
                memoryCell = (memoryCell + 1) % 256;
            else if (c == '.')
                output.Add(Convert.ToChar(memoryCell));
        }

        return string.Concat(output);
    }
}